elements/bulbs-page/bulbs-page.js

```diff
@@ -14,10 +14,27 @@ import invariant from 'invariant';
 
 import './bulbs-page.scss';
 
-// if we scroll past multiple pages at once they will all trigger a
-// 'pagestart' event. We only want to fire one. We use this debouncer
-// to skip all but the last 'pagestart' handler
-let pageStartDebouncer = debouncePerFrame();
+let inViewRectDebouncer = debouncePerFrame();
+let inViewQueue = [];
+
+function drainInViewQueue () {
```

Comment:
It seems sound to me.

We probably should get a confirmation from design that the urls are changing is a good place for UX.